ortools/linear_solver/xpress_interface.cc

```diff
@@ -1209,16 +1227,162 @@ void XpressInterface::SetLpAlgorithm(int value) {
   }
 }
 
+bool XpressInterface::readParameter(std::string const &name, std::string const &value)
+{
+  if (!value.size()) {
+    LOG(DFATAL) << "Empty value for parameter '" << name << "' in "
+                << SolverVersion();
+    return false;
+  }
+  int id, type;
+  if (XPRSgetcontrolinfo(mLp, name.c_str(), &id, &type) ||
+      type == XPRS_TYPE_NOTDEFINED)
+  {
+    LOG(DFATAL) << "Unknown parameter '" << name << "' in "
+                << SolverVersion();
+    return false;
+  }
+  std::stringstream v(value);
+  switch (type) {
+  case XPRS_TYPE_INT:
+  {
+    int i;
+    v >> i;
+    if (!v.eof()) {
+      LOG(DFATAL) << "Failed to parse value '" << value
+                  << "' for int parameter '" << name << "' in "
+                  << SolverVersion();
+      return false;
+    }
+    if (XPRSsetintcontrol(mLp, id, i)) {
+      LOG(DFATAL) << "Failed to set int parameter '" << name
+                  << "' to " << value << " (" << i << ") in "
+                  << SolverVersion();
+      return false;
+    }
+  }
+  break;
+  case XPRS_TYPE_INT64:
+  {
+    XPRSint64 i;
+    v >> i;
+    if (!v.eof()) {
+      LOG(DFATAL) << "Failed to parse value '" << value
+                  << "' for int64 parameter '" << name << "' in "
+                  << SolverVersion();
+      return false;
+    }
+    if (XPRSsetintcontrol64(mLp, id, i)) {
+      LOG(DFATAL) << "Failed to set int64 parameter '" << name
+                  << "' to " << value << " (" << i << ") in "
+                  << SolverVersion();
+      return false;
+    }
+  }
+  break;
+  case XPRS_TYPE_DOUBLE:
+  {
+    double d;
+    v >> d;
```

Comment:
I don't know what's under the hood of this `operator>>` for conversion. We noticed that `std::stod` had some issues with locales (e.g converting "8.4" to 8 if the locale was French).

Reply:
Good catch, thank you. I added `v.imbue(std::locale("C"));` in commit b4bc09bf73c86bc83e7dc45eb006ffc6dc9e5156 to make sure all parsing happens in the C locale.